esp/esp/program/modules/handlers/classsearchmodule.py

```diff
@@ -103,8 +103,14 @@ def classsearch(self, request, tl, one, two, module, extra, prog):
             'program': self.program,
             'query': None,
         }
+        namequery = request.GET.get('namequery')
+        decoded = None
         if data is not None:
             decoded = json.loads(data)
+        elif namequery: # only search if not None and not ""
+            decoded = {'filter': 'title', 'negated': False, 'values': [namequery]}
```

Comment:
Hmm, is there a reason not to do this client-side?  It seems like that leaves things more flexible if we want to change exactly what the query is later, or add another similar thing with a slightly different query.  Also just seems cleaner that way.